### PR TITLE
Added solution

### PR DESCRIPTION
**Pull request by AndrewPolishchuc**

### CODE REVIEW

**Review by andromash — APPROVED** [review]
Good job!

File: src/main/java/com/internet/shop/controllers/CartProductAddController.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.service.ProductService;
+import com.internet.shop.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class CartProductAddController extends HttpServlet {
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/CartProductAddController.java]:
```suggestion
public class AddProductToCartController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/CartProductAddController.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    private static final long USER_ID = 1;
+    private static final Injector INJECTOR = Injector.getInstance("com.internet.shop");
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/CartProductAddController.java]:
```suggestion
    private static final Injector injector = Injector.getInstance("com.internet.shop");
```

File: src/main/java/com/internet/shop/controllers/ProductAddController.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import com.internet.shop.lib.Injector;
+import com.internet.shop.model.Product;
+import com.internet.shop.service.ProductService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ProductAddController extends HttpServlet {
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/ProductAddController.java]: better AddProductController

File: src/main/java/com/internet/shop/controllers/ProductController.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import com.internet.shop.model.Product;
+import com.internet.shop.service.ProductService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ProductController extends HttpServlet {
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/ProductController.java]:
```suggestion
public class GetAllProductController extends HttpServlet {
```

**Review by Sofasmile — COMMENTED** [review]
good job

File: src/main/java/com/internet/shop/controllers/CartProductAddController.java (hunk 1 above)
Comment (AndrewPolishchuc) [reply src/main/java/com/internet/shop/controllers/CartProductAddController.java]: Sure. Done

File: src/main/java/com/internet/shop/controllers/ProductAddController.java (hunk 3 above)
Comment (AndrewPolishchuc) [reply src/main/java/com/internet/shop/controllers/ProductAddController.java]: Done)

File: src/main/java/com/internet/shop/controllers/ProductController.java (hunk 4 above)
Comment (AndrewPolishchuc) [reply src/main/java/com/internet/shop/controllers/ProductController.java]: Done.

File: src/main/java/com/internet/shop/controllers/CartProductAddController.java (hunk 2 above)
Comment (AndrewPolishchuc) [reply src/main/java/com/internet/shop/controllers/CartProductAddController.java]: Done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**